mypy.ini

```diff
@@ -308,6 +305,9 @@ ignore_missing_imports = True
 [mypy-prometheus_client.*]
 ignore_missing_imports = True
 
+[mypy-psycopg2.*]
+ignore_missing_imports = True
```

Comment:
No objection to ignoring this in config rather than in source code. However, typeshed maintains [stubs for psycopg2](https://pypi.org/project/types-psycopg2/). You could try pulling them in by adding them to `python_dependencies.py`. That might cause additional errors though; if so, feel free to abandon this idea for the future.

Reply:
My thought was that many exceptions are already defined in the central configuration. And why set one exception in the code and the other in the configuration. That is the reason for moving it conguration.
I had problems there: https://github.com/matrix-org/synapse/blob/c31da0cd92bf7faea1d4d04f967d057ad748cc1d/synapse/storage/databases/main/search.py#L277